### DISCUSSION TITLE
Best Practice for initializing/migrating new field to existing classes

### DISCUSSION THREAD

**Discussion by ErrolMatthiessen** [pimcore/pimcore#14621] [Q&A / Support]
Hi there,

I have often the case that I want to add a new field to a dataobject and generally I'm not sure how the best way is to initialize them.
So e.g. I want to add a new select field, which is required and has some default value.
After running pimcore:deployment:classes-rebuild the field gets created and gets initialize with null.

So I have to initialize them manually in some way, as the default value is not used for existing objects.
Mostly I used the batch update functionality in the list view, but then I had the case that I added two fields and then couldn't use it.

So how do you generally handle such a case?

Thanks for your support
Errol

**Comment by Dwarfex** [comment]
In my point of view we have two scenarios when creating new fields:

**Creating a new field with a default value**
Here we can easily distinguish the value we need to set -> the default 
Before reinventing the wheel - are we 100% sure the classes-rebuild command has no feature to apply the default?

We could create a command that walks all objects of a given class and sets the default if a property is NULL and a default is given
The runtime is depending on (existing objects of that class) multiplied by (properties of the class) which is a bit unfortunate.

Theoretically we could also check on loading an Object with doctrine if all required fields with defaults are completely filled ... 

 **Creating a new field without a default value**
 
 A command that sets a Field of a given Class to manual given value if the field is NULL
